File: examples/GatewayConnector.java
import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;
import java.time.Duration;

public class GatewayConnector {
    private final HttpClient client;
    private final String baseUrl;

    public GatewayConnector(String baseUrl) {
        this.baseUrl = baseUrl;
        this.client = HttpClient.newBuilder()
            .connectTimeout(Duration.ofSeconds(5))
            .build();
    }

    public int checkHealth() throws Exception {
        HttpRequest request = HttpRequest.newBuilder()
            .uri(URI.create(baseUrl + "/health"))
            .timeout(Duration.ofSeconds(5))
            .GET().build();
        return client.send(request, HttpResponse.BodyHandlers.ofString()).statusCode();
    }

    public static void main(String[] args) throws Exception {
        String url = args.length > 0 ? args[0] : "http://localhost:8080";
        int status = new GatewayConnector(url).checkHealth();
        System.out.println("Gateway " + url + " -> HTTP " + status);
    }
}

File: examples/GatewayConnector.kt
import java.net.URI
import java.net.http.HttpClient
import java.net.http.HttpRequest
import java.net.http.HttpResponse
import java.time.Duration

// OpenClaw Gateway Connector (Kotlin)
class GatewayConnector(private val baseUrl: String) {
    private val client: HttpClient = HttpClient.newBuilder()
        .connectTimeout(Duration.ofSeconds(5))
        .build()

    fun checkHealth(): Int {
        val request = HttpRequest.newBuilder()
            .uri(URI.create("$baseUrl/health"))
            .timeout(Duration.ofSeconds(5))
            .GET()
            .build()
        return client.send(request, HttpResponse.BodyHandlers.ofString()).statusCode()
    }
}

fun main(args: Array<String>) {
    val url = args.firstOrNull() ?: "http://localhost:8080"
    val status = GatewayConnector(url).checkHealth()
    println("Gateway $url -> HTTP $status")
}
